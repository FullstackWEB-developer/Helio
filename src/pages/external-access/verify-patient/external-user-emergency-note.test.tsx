import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {act, fireEvent, render, waitFor} from '@testing-library/react';
import TestWrapper from '../../../shared/test-utils/test-wrapper';
import ExternalUserCreateEmergencyNote from './external-user-emergency-note';
import { RequestChannel } from './models/request-channel.enum';
import { ExternalAccessRequestTypes } from '../models/external-updates-request-types.enum';
import Api from '@shared/services/api';
import { PatientExistsResponse } from '../models/patient-exists-response.model';
jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useParams: jest.fn().mockReturnValue({ type: "ticket-department" }),
  }))
  jest.setTimeout(70000);
describe("ExternalUserCreateEmergencyNote tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        callsLogState: {
            isFiltered: false
        },
        ticketState: {
            lookupValues: []
        },
        externalAccessState: {
            verifyPatientState: {
                redirectLink: {
                    attributes: {},
                    fullUrl: "string",
                    linkCreationDate: Date,
                    linkId: "string",
                    patientId: "string",
                    requestChannel: RequestChannel.Chat,
                    requestType: ExternalAccessRequestTypes.AppointmentConfirmation,
                    ticketId: "string",
                    sentAddress: "string"
                },
                phoneNumber: '',
                email: '',
                preventRetryUntil: 3,
                retryPrevented: false,
                twoFACodeResendDisabled: false,
                lastTwoFACodeSentTimestamp: undefined
            }
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders ExternalUserCreateEmergencyNote not-verified", async () => {
        let response = {
            data: {
                doesExists: true,
                email: "string",
                mobileNumber: "string",
                patientId: 75082
            } as PatientExistsResponse
        };
        jest.spyOn(Api, 'get').mockResolvedValue(response);
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <ExternalUserCreateEmergencyNote type={ExternalAccessRequestTypes.AppointmentList}/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders ExternalUserCreateEmergencyNote verified", async () => {
        let response = {
            data: {
                doesExists: true,
                email: "string",
                mobileNumber: "string",
                patientId: 75082
            } as PatientExistsResponse
        };
        jest.spyOn(Api, 'get').mockResolvedValue(response);
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <ExternalUserCreateEmergencyNote type={ExternalAccessRequestTypes.RequestRefill}/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });
})
