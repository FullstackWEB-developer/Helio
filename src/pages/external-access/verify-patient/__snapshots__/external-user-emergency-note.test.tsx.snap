// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalUserCreateEmergencyNote tests renders ExternalUserCreateEmergencyNote not-verified 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row space-x-6 items-center pt-16"
  >
    <div
      class=""
    >
      <svg
        class="icon-x-large "
        height="48"
        viewBox="0 0 48 48"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="48"
          width="48"
        />
        <path
          class="rgba-038-fill"
          d="M38,6H10a3.982,3.982,0,0,0-3.98,4L6,38a4,4,0,0,0,4,4H38a4,4,0,0,0,4-4V10A4,4,0,0,0,38,6ZM36,28H28v8H20V28H12V20h8V12h8v8h8Z"
        />
      </svg>
    </div>
    <div>
      external_access.emergency_note
    </div>
  </div>
</DocumentFragment>
`;

exports[`ExternalUserCreateEmergencyNote tests renders ExternalUserCreateEmergencyNote verified 1`] = `<DocumentFragment />`;
